ABSTRACT

De novo genome assembly has become one of the fundamental and computationally intensive tasks at hand. This is because of the nature of genomic sequences ranging from microorganism to higher organism. Thus it is very important to develop an efficient algorithmic approach, which could give us results using less resources and time. De Bruijn graph has proved to be a promising approach in the constructing the correct path and hence correct order of nucleotide. However with the ever-increasing in-flow of data from sequencing require more efficient data structure and algorithms. This report looks into the currently available assembly algorithms and implements the De Bruijn graph to find the correct order of nucleotide. Two data structure proposed by Ye et al. (2012) and Bowe et al. (2013) will be implemented in python. The results after implementation will be compared with ABySS, to test memory usage and correctness of implemented data structure.
